### PR TITLE
Log failed HTTP connections; include PID and TID in logs; include asset path in upload log messages

### LINKED ISSUE #416: even more logging detail ! ;)
```
                           SIZE        ERRORS     UPLOAD STATUS                MESSAGE
sub-YutaMouse33/sub-YutaMouse33_ses-YutaMouse33-150218_behavior+ecephys.nwb 11.9 GB        3              skipped               failed validation
sub-YutaMouse33/sub-YutaMouse33_ses-YutaMouse33-150223_behavior+ecephys.nwb 143.5 GB       0              ERROR                 ('Connection aborted.', OSError("(32, 'EPIPE')"))
```
but log has no detail:
```shell
drogon:~/.cache/dandi-cli/log
$> grep -i aborted 20210222181008Z-6322.log
```
I guess is happening higher in the overall `upload` command.

unrelated: since we are uploading in parallel (iirc threads) may be worth adding by default a thread id into each log line, but iirc we also have multiprocessing used, so we better add both - `PID:THREAD` in each line.  Otherwise it is hard to impossible to associate different log lines to different ongoing processes/files  e.g. in 
```
2021-02-22T13:10:11-0500 [DEBUG   ] dandi Calculated sha256 digest of 81a6e764bda0c0cfa4b96d200b103d347e7d7c7c4ee82665b5b9784f8d2084cf for sub-YutaMouse33/sub-YutaMouse33_ses-YutaMouse33-150226_behavior+ecephys.nwb
2021-02-22T13:10:11-0500 [DEBUG   ] dandi POST https://api.dandiarchive.org/api/uploads/validate/
2021-02-22T13:10:11-0500 [DEBUG   ] dandi Response: 400
2021-02-22T13:10:11-0500 [ERROR   ] dandi Error 400 while sending POST request to https://api.dandiarchive.org/api/uploads/validate/: ["A validation for an object with that checksum does not exist."]
2021-02-22T13:10:11-0500 [DEBUG   ] dandi Blob does not already exist on server
2021-02-22T13:10:11-0500 [DEBUG   ] dandi Beginning upload
2021-02-22T13:10:11-0500 [DEBUG   ] dandi POST https://api.dandiarchive.org/api/uploads/initialize/
2021-02-22T13:10:12-0500 [DEBUG   ] dandi Response: 204
2021-02-22T13:10:12-0500 [DEBUG   ] dandi Blob is already uploaded to server
2021-02-22T13:10:12-0500 [DEBUG   ] dandi Waiting for server-side validation to complete
2021-02-22T13:10:12-0500 [DEBUG   ] dandi GET https://api.dandiarchive.org/api/uploads/validations/81a6e764bda0c0cfa4b96d200b103d347e7d7c7c4ee82665b5b9784f8d2084cf/
2021-02-22T13:10:12-0500 [DEBUG   ] dandi Response: 200
```

### PR DESCRIPTION
**Pull request by jwodder**
Closes #416.

Note that the process and thread ID are only included in the logs logged to the log directory.  Let me know if you want them included in the logs printed to the screen as well.

### CODE REVIEW

File: dandi/dandiapi.py [hunk 1]
```diff
@@ -132,15 +132,20 @@ def send_request(
...
+            result = f(
+                url,
+                params=parameters,
+                data=data,
+                files=files,
+                json=json,
+                headers=_headers,
+                **kwargs,
+            )
+        except Exception:
```
Comment (yarikoptic) [inline dandi/dandiapi.py:145]:
```suggestion
        except Exception as exc:
```

File: dandi/dandiapi.py [hunk 2]
```diff
@@ -132,15 +132,20 @@ def send_request(
...
+            lgr.exception("HTTP connection failed")
```
Comment (yarikoptic) [inline dandi/dandiapi.py:146]:
```suggestion
            lgr.exception("HTTP connection failed: %s", exc)
```
so we log how it failed

**Review by yarikoptic — CHANGES_REQUESTED** [review]
A little more information suggested.  I am ok to log pid:tid only in the log file.

File: dandi/dandiapi.py (hunk 2 above)
Comment (jwodder) [reply dandi/dandiapi.py:146]: [The `logging.exception()` method](https://docs.python.org/3/library/logging.html#logging.Logger.exception) already adds exception information.

Comment (yarikoptic) [reply dandi/dandiapi.py:146]: oh, never used it. Good to know, thanks!

**Pull request merged**